File: ccc/2015/s3.py
file = open("/home/wylited/Documents/coding/Competitive Programming/ccc/2015/s3.9.in")

ports = int(file.readline())
numPlanes = int(file.readline())

planes = []
total = 0

for i in range(ports + 1):
    planes.append(0)

j = 0
planesLeft = True
while j < numPlanes and planesLeft:
    incPlane = int(file.readline()) # read in the next plane

    while incPlane > 0 and planes[incPlane] > 0: # if plane is already in the port
        cport = planes[incPlane] # get the port it's in
        planes[incPlane] = planes[incPlane] + 1 # increment the port
        incPlane = incPlane - cport # move the plane to the next port

    if incPlane <= 0:
        planesLeft = False # if no planes left, stop

    else:
        planes[incPlane] = 1 # add plane to port
        total = total + 1 # if plane is added, add to total

    j = j + 1

print(total)

File: ccc/2015/s3.cpp
#include <fstream>
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int ports = 0, numPlanes = 0;
    vector<int> planes(10000, 0);
    ifstream file;
    file.open("/home/wylited/Documents/coding/Competitive Programming/ccc/2015/s3.9.in");

    file >> ports >> numPlanes;

    for (int i = 0; i < numPlanes; i++) {
        file >> planes[i];
    }

    int j = 0, cplane = 0, cport = 0, total = 0;
    bool planesLeft = true;
    while(j < numPlanes && planesLeft){
        file >> cplane;
        while (cplane > 0 && planes[cplane] > 0){
            cport = planes[cplane];
            ++planes[cplane];
            cplane = cplane - cport;
        }
        if (cplane <= 0) {
            planesLeft = false;
        } else {
            planes[cplane] = 1;
            ++total;
        }
        ++j;
    }
    return 0;
}